### PR TITLE
Changing the behavior for ray tracing in case a submesh has both opaque and transparent sub meshes

### PR DESCRIPTION
**Pull request by anisunity**
All in the title, like we discussed

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassRaytracingGBuffer.hlsl [hunk 1]
```diff
@@ -48,6 +48,9 @@ void ClosestHitGBuffer(inout RayIntersectionGBuffer rayIntersectionGbuffer : SV_
 [shader("anyhit")]
 void AnyHitGBuffer(inout RayIntersectionGBuffer rayIntersectionGbuffer : SV_RayPayload, AttributeData attributeData : SV_IntersectionAttributes)
 {
+#ifdef _SURFACE_TYPE_TRANSPARENT
+    IgnoreHit();
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassRaytracingGBuffer.hlsl]: Is it possible to just not declare the AnyHitGBuffer at all with the #ifdef?

**Comment by anisunity** [comment]
@sebastienlagarde is your question the one we discussed?

**Comment by anisunity** [comment]
A use case example would be to take a mesh renderer with multiple submeshes some of them transparent and some of them opaque. Enable ray traced reflection on a mirror an observe what appears/disappears with this change. It would be also good to see how the behavior changes as all the objects become transparent or opaque.

**Comment by anisunity** [comment]
the bug reported by @remi-chapelain happening when the number of materials is lower than the number of submeshes has been included in this PR

**Pull request merged**